src/core/meshcop/dtls.cpp

```diff
@@ -285,7 +286,17 @@ otError Dtls::Setup(bool aClient)
     mbedtls_ssl_conf_max_version(&mConf, MBEDTLS_SSL_MAJOR_VERSION_3, MBEDTLS_SSL_MINOR_VERSION_3);
 
     OT_ASSERT(mCipherSuites[1] == 0);
+    OT_ASSERT(mCurves[1] == MBEDTLS_ECP_DP_NONE);
     mbedtls_ssl_conf_ciphersuites(&mConf, mCipherSuites);
+    mbedtls_ssl_conf_curves(&mConf, mCurves);
+#if defined(MBEDTLS_KEY_EXCHANGE__WITH_CERT__ENABLED)
+    if (mCipherSuites[0] == MBEDTLS_TLS_ECJPAKE_WITH_AES_128_CCM_8)
+    {
+        int hashes[1];
```

Comment:
From the implementation of mbedtls_ssl_conf_sig_hashes(), it seems it cannot accept a list in a stack.

Reply:
@bukepo you are right, I'll add mHashes to Dtls class.

And I guess it would be nice to restrict the SHAs for ECDSA as well, but not sure which are required...

Reply:
I think these restrictions should only be applied to the dtls socket used for Thread commissioning. Could we keep the base Dtls not restricted? Maybe we can move the current `ot::MeshCoP::Dtls` to `ot::Net::Dtls`, and the `ot::MeshCoP::Dtls` inherits the basic one and adds these restrictions. Thoughts?

Reply:
@bukepo I would say that makes sense, however it's a bigger change.

By the moment I'll redo this PR just to match its description regarding ECJPAKE only.